### PR TITLE
gh-116991: Improve `pygen --help` for `python` subparser

### PR DESCRIPTION
**Pull request by sobolevn**
* Issue: gh-116991

### CODE REVIEW

File: Tools/peg_generator/pegen/__main__.py [hunk 1]
```diff
@@ -107,7 +107,13 @@ def generate_python_code(
     help="Suppress code emission for rule actions",
 )
 
-python_parser = subparsers.add_parser("python", help="Generate Python code")
+python_parser = subparsers.add_parser(
+    "python",
+    help=(
+        "Generate Python code, needs grammar definition with Python actions, "
```
Comment (vasily-v-ryabov) [inline Tools/peg_generator/pegen/__main__.py:113]: I would say `needs Python-specific grammar file (different from cpython/Grammar/python.gram which is for parser in C), `

**Review by vasily-v-ryabov — COMMENTED** [review]
Thanks for improving it and for the help! I personally started from CPython repository which contains folder `Grammar` with only `python.gram`. It would be nice to have `python_in_c.gram` and `python_in_python.gram` which makes the picture clear from the very beginning. But this is only my path. Maybe other paths could be improved as well. I didn't see it is well highlighted in Dev Guide.

**Review by lysnikolaou — APPROVED** [review]
LGTM

File: Tools/peg_generator/pegen/__main__.py [hunk 2]
```diff
@@ -107,7 +107,13 @@ def generate_python_code(
...
+        "see `we-like-parsers/pegen`"
```
Comment (vasily-v-ryabov) [inline Tools/peg_generator/pegen/__main__.py]: I'd suggest more details at this line:
```
see repository `we-like-parsers/pegen`, file `data/python.gram` for example.
```

**Review by vasily-v-ryabov — APPROVED** [review]
LGTM, thanks!

File: Tools/peg_generator/pegen/__main__.py [hunk 3]
```diff
@@ -107,7 +107,13 @@ def generate_python_code(
     help="Suppress code emission for rule actions",
 )
 
-python_parser = subparsers.add_parser("python", help="Generate Python code")
+python_parser = subparsers.add_parser(
+    "python",
+    help=(
+        "Generate Python code, needs grammar definition with Python actions, "
+        "see `we-like-parsers/pegen`, start with `data/python.gram`"
```
Comment (pablogsal) [inline Tools/peg_generator/pegen/__main__.py]: I don't think we should mention that repo. The input for the parser can be any grammar other than the general Python one. We use to generate the meta parser for example. I think linking to another repo it's actually more confusing.

This is also an internal tool without external user guarantees.

I would be on improving the help to mention a bit of this but I still prefer not to link against the original repo

Comment (sobolevn) [reply Tools/peg_generator/pegen/__main__.py]: Ok, how about this?

**Comment by sobolevn** [comment]
I've merged a simple version of `--help` update, without the link to the external repo.
Thanks everyone!

**Pull request merged**